pkg/bgpv1/manager/reconcilerv2/crd_status.go

```diff
@@ -0,0 +1,277 @@
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	k8s_client "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	CRDStatusUpdateInterval = 5 * time.Second
+)
+
+type StatusReconciler struct {
+	lock.Mutex
+
+	Logger            logrus.FieldLogger
+	ClientSet         k8s_client.Clientset
+	LocalNodeResource daemon_k8s.LocalCiliumNodeResource
+
+	nodeName      string
+	desiredStatus *v2alpha1.CiliumBGPNodeStatus
+	runningStatus *v2alpha1.CiliumBGPNodeStatus
+}
+
+type StatusReconcilerIn struct {
+	cell.In
+
+	Job       job.Group
+	ClientSet k8s_client.Clientset
+	Logger    logrus.FieldLogger
+	LocalNode daemon_k8s.LocalCiliumNodeResource
+}
+
+type StatusReconcilerOut struct {
+	cell.Out
+
+	Reconciler StateReconciler `group:"bgp-state-reconciler"`
+}
+
+func NewStatusReconciler(in StatusReconcilerIn) StatusReconcilerOut {
+	// CRD Status reconciler is disabled if there is no kubernetes support
+	if !in.ClientSet.IsEnabled() {
+		return StatusReconcilerOut{}
+	}
+
+	r := &StatusReconciler{
+		Logger:            in.Logger.WithField(types.ReconcilerLogField, "CRD_Status"),
+		LocalNodeResource: in.LocalNode,
+		ClientSet:         in.ClientSet,
+	}
+
+	in.Job.Add(job.OneShot("bgp-crd-status-initialize", func(ctx context.Context, health cell.Health) error {
+		r.Logger.Debug("Initializing")
+
+		for event := range r.LocalNodeResource.Events(ctx) {
+			switch event.Kind {
+			case resource.Upsert:
+				r.Lock()
+				r.nodeName = event.Object.GetName()
+				r.Unlock()
+			}
+			event.Done(nil)
+		}
+		return nil
+	}))
+
+	in.Job.Add(job.OneShot("bgp-crd-status-update-job", func(ctx context.Context, health cell.Health) (err error) {
+		r.Logger.Debug("Update job running")
+
+		ticker := time.NewTicker(CRDStatusUpdateInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				err := r.reconcileCRDStatus(ctx)
+				if err != nil {
+					r.Logger.WithError(err).Error("Failed to update CRD status")
+				}
+
+			case <-ctx.Done():
+				r.Logger.Debug("CRD status update job stopped")
+				return nil
+			}
+		}
+	}))
+
+	return StatusReconcilerOut{
+		Reconciler: r,
+	}
+}
+
+func (r *StatusReconciler) Name() string {
+	return "CRDStatus"
+}
+
+func (r *StatusReconciler) Priority() int {
+	return 50
+}
+
+func (r *StatusReconciler) Reconcile(ctx context.Context, params StateReconcileParams) error {
+	r.Lock()
+	defer r.Unlock()
+
+	r.Logger.WithFields(logrus.Fields{
+		types.InstanceLogField: params.UpdatedInstance.Config.Name,
+	}).Debug("Reconciling CRD status")
+
+	// do not reconcile if not in BGPv2 mode
+	if params.ConfigMode.Get() != mode.BGPv2 {
+		return nil
+	}
+
+	current := &v2alpha1.CiliumBGPNodeStatus{}
+	if r.desiredStatus != nil {
+		current = r.desiredStatus.DeepCopy()
+	}
```

Comment:
We can simply initialize r.desiredStatus with `&v2alpha1.CiliumBGPNodeStatus{}` in the constructor and change this to `current = r.desiredStatus.DeepCopy()`.